modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java

```diff
@@ -89,7 +92,8 @@ public void testPrecisionAtRequest() {
 
         RankEvalResponse response = client().execute(RankEvalAction.INSTANCE, builder.request())
                 .actionGet();
-        assertEquals(1.0, response.getEvaluationResult(), Double.MIN_VALUE);
+        double expectedPrecision = (1.0 / 6.0 + 4.0 / 6.0) / 2.0;
```

Comment:
@cbuescher Can you please explain what is this calculation doing  `(1.0 / 6.0 + 4.0 / 6.0) / 2.0`?  The way I understood PrecisionAtK:  for this request we've got 6 docs, and 4 of the relevant, so according to the formula `TP/(TP+FP)`, shouldn't we have `4/(4+2)`?  Or we are calculation some other type of precision here?

Reply:
Good thing you ask, I should really add some comment here, also had to think a little about this numbers. We use the `ignoreUnlabeled == false` option in this test, and there are two queries.
We retrieve all six documents for both queries, the "berlin" query has one relevant document, the five unlabeled ones are treated as unrelevant, so precision is 1/6 for this query. For the other there are four relevant docs, so precion is 4/6. Currently we average over the precision values for each query (information need) to make this metric similar in behaviour to MRR and NDCG, but I just checked in the literature and couldn't find so much mentionings of this. I'm not sure if there are any other good ways of combining results for Prec@ across multiple queries though.
Another thing to note: if the number of documents returned is less than k (here we calculate Prec@10 but have only 6 documents) we don't penalize for the missing documents (which is sometimes done and sometimes not in the literature I found). This might be another esage case that we need to consider changing or making configurable, but we wanted to go with a simple version first. I will see if I can add some comment on this to the test.

Reply:
@cbuescher thanks very much for your detailed explanation. I have not noticed that there was another berlin query. It makes total sense to average precision across queries.

Reply:
np, its good to rethink these calculations over and over, as I said above I'm not entirely sure about all the edge cases, would like to have a chat about this maybe on the main PR. I added 8e5286118545d78deeccf06d76a4f428e8e3f59f to #27595 that hopefully make the existing tests a little bit clearer. Feel free to leave your comments on that PR.